### PR TITLE
WFCORE-2021 informative error message if it can not convert result to…

### PR DESCRIPTION
**Pull request by soul2zimate**
… list.
https://issues.jboss.org/browse/WFCORE-2021

### CODE REVIEW

**Comment by bstansberry** [comment]
@jfdenise can confirm but I think the problem here is that the ls command is not designed to handle wildcard addresses (i.e. those with * in the address). Which I think is fine. But I think the solution then should be to detect such addresses and reject them before submission to the server, with a targeted error message.

The fact that the subsystem=* example in the JIRA fails with a somewhat meaningful error message from the server side is really just luck. It fails because it is trying to read resource description data for a resource type (subsystem=*) that has no registration. There is no wildcard resource registration for "subsystem" while there is for "socket-binding-group."

**Comment by soul2zimate** [comment]
Yes, I agree with you. I will see how to distinguish such addresses

**Comment by jfdenise** [comment]
ls handler is not able to deal with response being a list of resources (as @soul2zimate detailed in the bug description). It could be enhanced but rejecting the syntax seems a good fix easy to change in case we want to support it.

**Comment by soul2zimate** [comment]
I have rebased this to reject any address contains * in ls handler before sending to server

**Comment by jfdenise** [comment]
Hi, I would possibly replace the message by "* not supported in node path argument"

**Comment by soul2zimate** [comment]
Thanks, the message is updated as suggested.

**Comment by soul2zimate** [comment]
retest this please

File: cli/src/main/java/org/jboss/as/cli/handlers/LsHandler.java [hunk 1]
```diff
@@ -571,6 +571,9 @@ private ModelNode getAddressNode(CommandContext ctx, OperationRequestAddress add
             while (iterator.hasNext()) {
                 OperationRequestAddress.Node node = iterator.next();
                 if (node.getName() != null) {
+                    if (node.getName().equals("*")) {
+                        throw new CommandFormatException("* is not supported in node path argument");
```
Comment (bstansberry) [inline cli/src/main/java/org/jboss/as/cli/handlers/LsHandler.java]: @jfdenise Do we use the 'node' much in CLI docs, help, etc?

I think I've seen it before but it's a bit odd to use that term  in messages unless it's the typical way navigation is described.

Comment (jfdenise) [inline cli/src/main/java/org/jboss/as/cli/handlers/LsHandler.java]: @bstansberry , yes node path is used specially in the ls --help.

**Pull request merged**